pkg/apis/gcp/validation/infrastructure_test.go

```diff
@@ -293,16 +301,59 @@ var _ = Describe("InfrastructureConfig validation", func() {
 			Expect(ValidateInfrastructureConfigUpdate(infrastructureConfig, infrastructureConfig, fldPath)).To(BeEmpty())
 		})
 
-		It("should forbid changing the network section", func() {
+		It("should allow changing cloudNAT details", func() {
```

Comment:
Can you also add a test case that checks if changes to the flow logs are allowed? Or, do it like with the negative test, change this one here to try changing all the fields that should be allowed to change